### ISSUE TITLE
 [UPGRADE] Upgrade PostgreSQL to v13

### ISSUE THREAD

**Issue by rafzei** [epiphany-platform/epiphany#1861]
**Is your feature request related to a problem? Please describe.**
- PostgreSQL v10 has 20 vulnerabilities : [CVE-2019-10211, CVE-2018-16850, CVE-2018-1115, CVE-2019-10208, CVE-2019-10164, CVE-2018-1058, CVE-2018-10925, CVE-2018-10915, CVE-2020-10733, CVE-2020-14350,CVE-2019-9193, 
CVE-2020-14349, CVE-2019-10210, CVE-2018-1053, CVE-2018-1052, CVE-2017-15099, CVE-2020-1720, CVE-2019-10130](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&seach_type=all&query=cpe:2.3:a:postgresql:postgresql:10.0:*:*:*:*:*:*:*)
- PostgreSQL v12 have also additional functional, performance and security improvements comparing to PostgreSQL v10

**Describe the solution you'd like**
- Upgrade PostgreSQL to version 13
- Backport to supported Epiphany versions (please consider creating new issues for this purpose)

**Describe alternatives you've considered**
None

**Additional context**
External components related to PostgreSQL should be also upgraded: Rep Mgr, PGAudit, pgpool, PgBouncer. There are separate tasks created for those upgrades. We also need to check possibility of this upgrade in context of our components used for replication and if those components are still needed(can be replaced by PostgreSQL embedded or other components functionality).

---

**DoD checklist**

* [ ] Changelog updated
* [ ] COMPONENTS.md updated / doesn't need to be updated
* [ ] Feature has automated tests
* [ ] Automated tests passed (QA pipelines)
  * [ ] apply
  * [ ] upgrade
* [ ] Idempotency tested
* [ ] Documentation added / updated / doesn't need to be updated
* [ ] All conversations in PR resolved
* [ ] Solution meets requirements and is done according to design doc
* [ ] Usage compliant with license

**Comment by atsikham** [comment]
* After Postgres upgrading related components have to be checked: `repmgr, pgpool, pgbouncer, pgaudit`. So this task may be large, especially with testing part.
* Let's discuss backporting tasks after implementation, during DoD.

**Comment by mkyc** [comment]
we should split in 3 tasks: 
 - installation
 - spike about upgrade
 - upgrade

**Comment by mkyc** [comment]
Also ensure that we have all for ARM architecture before we start here.

**Comment by mkyc** [comment]
Also start with checking that we can upgrade PostgreSQL and it's surrounding would be compatible.

**Comment by atsikham** [comment]
Additional links:

* https://www.postgresql.org/docs/13/pgupgrade.html
* https://repmgr.org/docs/current/upgrading-repmgr-extension.html

**Comment by atsikham** [comment]
@erzetpe @mkyc could you clarify if we are going to support upgrade of postgresql or only apply.

**Comment by erzetpe** [comment]
In my opinion on every environment where the data has some business value the upgrade should be performed manually with someone that is experienced Postgres DBA.

**Comment by atsikham** [comment]
Decision:

* do upgrades automatically
* ask user a question about backup before starting upgrade
* check if there is a "dry run" option for upgrade. If yes, add it to preflight

**Comment by mkyc** [comment]
if there is no "dry run" perform queries @to-bar mentioned, to check that unsupported types of data types do not exist in DB.

**Comment by atsikham** [comment]
Merged to feature/upgrade-postgresql-13 branch, but moved to 'on hold' as waiting for other postgres parts to be merged.

**Comment by przemyslavic** [comment]
NEW DEPLOYMENTS:
:heavy_check_mark: single machine: `apply`, re-`apply`
:heavy_check_mark: single node: `apply`, re-`apply`
:heavy_check_mark: single node: `apply`, re-`apply` - aarch64
:heavy_check_mark: master + replica: `apply`, re-`apply`
:heavy_check_mark: master + 2 replicas: `apply`, re-`apply`
:heavy_check_mark: repmgr standby switchover
:heavy_check_mark: automatic failover

UPGRADES:
:heavy_check_mark: `upgrade` from v0.8 to dev: 1 postgresql node with sample data inserted before executing the upgrade command.
:heavy_check_mark: `upgrade` from v0.8 to dev: 2 postgresql nodes (clustered) with sample data inserted before executing the upgrade command + reapply.
:heavy_check_mark: `upgrade` from v0.9 to dev:  1 postgresql node (single machine) with sample data inserted before executing the upgrade command
:heavy_check_mark: `upgrade` from v0.9 to dev: 2 postgresql nodes (clustered) with sample data inserted before executing the upgrade command + reapply.
:heavy_check_mark: `upgrade` from v1.0 to dev: 1 postgresql node with sample data inserted before executing the upgrade command + reapply.
:heavy_check_mark: `upgrade` from v1.0 to dev: 2 postgresql nodes (clustered) with sample data inserted before executing the upgrade command + reapply.
:heavy_check_mark: `upgrade` from v1.1 to dev: 2 postgresql nodes (clustered) with sample data inserted before executing the upgrade command + reapply.
:heavy_check_mark: `upgrade` from v1.1 to dev: 1 postgresql node + reapply (aarch64)

===================================================================

TO BE FIXED:
#2573
#2565
#2562
#2578
#2583

TO BE TESTED:
#2498

These issues will be tracked in specific tasks.
